src/frontend/src/icp/services/icrc.services.ts

```diff
@@ -171,7 +165,7 @@ const loadCustomIcrcTokensData = async ({
 				? meta
 				: await metadata({ ledgerCanisterId: ledgerCanisterIdText, identity, certified }),
 			ledgerCanisterId: ledgerCanisterIdText,
-			indexCanisterId: indexCanisterId.toText(),
+			...(nonNullish(indexCanisterId) && { indexCanisterId: indexCanisterId.toText() }),
```

Comment:
If you rather like to have `indexCanisterId: undefined` within the memory not against. Instead of

```
			...(nonNullish(indexCanisterId) && { indexCanisterId: indexCanisterId.toText() }),
```

we can also do

```
indexCanisterId: indexCanisterId?.toText()
```

As you rather like Antonio.

Reply:
nono, i like it like this, so that the index canister prop is really missing and not `undefined`